subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/attributes/ClasspathDependenciesAttributesIntegrationTest.groovy

```diff
@@ -21,12 +21,84 @@ import org.gradle.integtests.fixtures.RequiredFeature
 import org.gradle.integtests.fixtures.RequiredFeatures
 import org.gradle.integtests.resolve.AbstractModuleDependencyResolveTest
 import org.gradle.test.fixtures.file.TestFile
+import org.gradle.test.fixtures.plugin.PluginBuilder
 import org.objectweb.asm.ClassWriter
 import org.objectweb.asm.Opcodes
 import org.objectweb.asm.tree.ClassNode;
 
 class ClasspathDependenciesAttributesIntegrationTest extends AbstractModuleDependencyResolveTest {
 
+    def pluginBuilder = new PluginBuilder(file('plugin'))
+
+    @RequiredFeatures([
+        @RequiredFeature(feature = GradleMetadataResolveRunner.GRADLE_METADATA, value = "false"),
+        @RequiredFeature(feature = GradleMetadataResolveRunner.EXPERIMENTAL_RESOLVE_BEHAVIOR, value = "false"),
+        @RequiredFeature(feature = GradleMetadataResolveRunner.REPOSITORY_TYPE, value = "maven")
+    ])
+    def 'plugin settings useModule derives variants'() {
+        given:
+        def module = mavenRepo.module('test', 'dep', '1.0').publish()
+        mavenRepo.module('test', 'bom', '1.0').hasPackaging('pom').dependencyConstraint(module).publish()
+
+        pluginBuilder.addPlugin("println 'hello'")
+        def pluginModule = pluginBuilder.publishAs('test:plugin:1.0', mavenRepo, executer).pluginModule
+        def pomFile = pluginModule.pom.file
+        pomFile.text = pomFile.text.replace('</project>', """
```

Comment:
Not too excited about this trick, but the `PluginBuilder` would need deeper change to support having dependencies in plugins it creates.